torchaudio/models/decoder/_ctc_decoder.py

```diff
@@ -270,12 +282,22 @@ def ctc_decoder(
         >>> )
         >>> results = decoder(emissions) # List of shape (B, nbest) of Hypotheses
     """
+    if lm_dict is not None and type(lm_dict) is not str:
+        raise ValueError("lm_dict must be None or str type.")
+
     tokens_dict = _Dictionary(tokens)
+    if lm_dict:
+        word_dict = _Dictionary(lm_dict)
```

Comment:
super nit, but the lexicon is also a dictionary of words, so maybe call this `_lm_dict`

Reply:
I have it as `word_dict` (lexicon dict) in the current implementation because if `lm_dict` is provided, it actually is used instead of the lexicon file to construct the dictionary of words. this way the custom LM index and scoring output aligns with the lexicon indexing, and we assume lm_dict and lexicon consist of the same words. let me know if this is reasonable

Reply:
Hmm — I see what you're saying. This could work if that's what you want the behavior to be, but I'm not sure if this behavior makes sense.

First — does this work as expected? The LM dict is typically a mapping of words to ID — this needs to be parsed out so it's only a list of words, right? There are a few other assumptions here: first, if you're constructing that dictionary manually, it assumes you're doing token-level decoding. You'd need to check consistency between the token set in the LM dict and the token output from the model.

Thing is, we've done lexicon free training with language models, so you still might not want a lexicon: https://github.com/flashlight/wav2letter/tree/main/recipes/lexicon_free